test/SparqlLexerTest.cpp

```diff
@@ -188,8 +188,20 @@ TEST(SparqlLexer, peek) {
 
   SparqlLexer lexer(query);
   ASSERT_TRUE(lexer.peek("prefix"));
+  ASSERT_FALSE(lexer.peek("select"));
   ASSERT_EQ(std::string("prefix"), lexer.getUnconsumedInput());
 
   ASSERT_TRUE(lexer.peek(SparqlToken::Type::KEYWORD));
+  ASSERT_FALSE(lexer.peek(SparqlToken::Type::GROUP_BY));
   ASSERT_EQ(std::string("prefix"), lexer.getUnconsumedInput());
+
+  // Test with RDF_LITERAL because these are not lowercased by the lexer
+  std::string query2 = R"("RDF_LITERAL")";
+  SparqlLexer lexer2(query2);
+  ASSERT_TRUE(lexer2.peek(SparqlToken::Type::RDFLITERAL));
+  ASSERT_TRUE(lexer2.peek("\"RDF_LITERAL\""));
+  ASSERT_TRUE(lexer2.peek("\"RDF_LITERAL\"", true));
+  ASSERT_FALSE(lexer2.peek("\"rdf_literal\"", true));
```

Comment:
If you have two test cases like this then either
- wrap them in different scopes
```
{
  auto query = ...
   auot lexer = ...
}
{
  auto query = ...
  auto lexer = ...
}
or even set up to test cases
```
TEST(SparqlLexer, PeekKeeword) {...}
TEST(SparqlLexer, PeekLiteral) {...}
```
This makes it clear that none of the variables from either case leak into the other and reduces the need of `something2` variables because there is a new scope.

Reply:
Will do.

Do you have any Naming conventions for tests etc.? I saw that same Tests are named Camel Case while others are Pascal Case (=Camel Case with uppercase first letter).

Reply:
Thanks for pointing this out.
Let it hereby be known, that we use PascalCase for TestCaseNames
(TestName.TestCaseName is easier to parse than TestName.testCaseName for a human).
And that all other test cases have wrong names and should be updated.